resolwe_bio/processes/scseq/cellranger.py

```diff
@@ -0,0 +1,260 @@
+        matrix_raw = FileField(label='Matrix (raw)')
+        genes_raw = FileField(label='Genes (raw)')
+        barcodes_raw = FileField(label='Barcodes (raw)')
+        report = FileHtmlField(label="Report")
+        build = StringField(label="Build")
+        species = StringField(label="Species")
+        source = StringField(label="Gene ID source")
+
+    def run(self, inputs, outputs):
+        sample_name = inputs.reads.entity_name.strip('.fastq.gz')
+
+        dir_fastqs = './fastqs'
+        os.mkdir(dir_fastqs)
+
+        # Format cellranger count fastq input so it follows the correct naming convention and
+        # folder structure
+        for i, fastqs in enumerate(zip(inputs.reads.barcodes, inputs.reads.reads)):
+            os.symlink(
+                fastqs[0].path,
+                os.path.join(
+                    dir_fastqs,
+                    '{}_S1_L{}_R1_001.fastq.gz'.format(
+                        sample_name,
+                        str(i+1).zfill(3),
+                    )
+                )
+            )
+            os.symlink(
+                fastqs[1].path,
+                os.path.join(
+                    dir_fastqs,
+                    '{}_S1_L{}_R2_001.fastq.gz'.format(
+                        sample_name,
+                        str(i+1).zfill(3),
+                    )
+                )
+            )
+
+        cmd = Cmd['cellranger']['count']
+        cmd = cmd['--id={}'.format(sample_name)]
+        cmd = cmd['--fastqs={}'.format(dir_fastqs)]
+        cmd = cmd['--transcriptome={}'.format(inputs.genome_index.genome_index.path)]
+        cmd = cmd['--localcores={}'.format(self.requirements.resources.cores)]
+        cmd = cmd['--localmem={}'.format(int(self.requirements.resources.memory * 0.9 / 1024))]
+        cmd = cmd['--chemistry={}'.format(inputs.chemistry)]
+        cmd = cmd['--expect-cells={}'.format(inputs.expected_cells)]
+        if inputs.trim_r1:
+            cmd = cmd['--r1-length={}'.format(inputs.trim_r1)]
+        if inputs.trim_r2:
+            cmd = cmd['--r2-length={}'.format(inputs.trim_r2)]
+        if inputs.force_cells:
+            cmd = cmd['--force-cells={}'.format(inputs.force_cells)]
+        cmd()
+
+        output_dir = '{}/outs'.format(sample_name)
+
+        # Spawn upload-bam process
+        bam_name = '{}.bam'.format(sample_name)
+        bai_name = '{}.bam.bai'.format(sample_name)
+        bam_path = os.path.join(output_dir, 'possorted_genome_bam.bam')
+        bai_path = os.path.join(output_dir, 'possorted_genome_bam.bam.bai')
+        os.symlink(bam_path, bam_name)
+        os.symlink(bai_path, bai_name)
+        process_inputs = {
+            'src': bam_name,
+            'src2': bai_name,
+            'reads': inputs.reads.id,
+            'species': inputs.genome_index.species,
+            'build': inputs.genome_index.build,
+        }
+        self.run_process('upload-bam-scseq-indexed', process_inputs)
```

Comment:
The process does not get spawned and I do not know why. It works if I run it by itself.